### DISCUSSION TITLE
are function names only available in the stacktrace when the program is compiled with -g?

### DISCUSSION THREAD

**Discussion by marlowa** [jeremy-rifkin/cpptrace#142] [Q&A]
Hello everyone,

My experiments with cpptrace seem to indicate that you don't get function names in the stacktrace unless the program is compiled with -g. Is that right? When I use gdb on a program compiled without -g I can still breakpoint on named function entry and the stacktrace does include the function name.
If this is a limitation where does it come from? cpptrace? Or maybe libdwarf? Is it documented anywhere?

**Comment by jeremy-rifkin** [comment]
Hi, yes. The library relies on debug symbols to provide both symbol info and line info, the need to use -g is documented in the README. Gcc is likely showing the function names by looking them up in the binary's symbol table or dynamic symbol table. It would be possible for cpptrace to do this as well, however, there have been other aspects of the library to prioritize.
